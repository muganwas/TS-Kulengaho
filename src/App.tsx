import React from 'react'
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from './redux/store'
import Home from './components/Home'
import Dash from './components/Dashboard'

const App: React.FC = () => {
	return (
		<Provider store={store}>
			<BrowserRouter basename='/'>
				<Switch>
					<Route exact path='/' component={Home} />
					<Route exact path='/Dashboard' component={Dash} />
				</Switch>
			</BrowserRouter>
		</Provider>
	)
}

export default App
