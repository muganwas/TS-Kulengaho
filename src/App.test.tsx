import React from 'react'
import { render } from '@testing-library/react'
import App from './App'

describe('Landing page renders', () => {
  test('renders sub-title', () => {
    const { getByText } = render(<App />)
    const subtitle = getByText(/kulengaho/i)
    expect(subtitle).toBeInTheDocument()
  })
})
