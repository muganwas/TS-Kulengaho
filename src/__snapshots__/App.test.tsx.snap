// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page renders snapshot checksout 1`] = `
<div
  className="home-container"
  data-testid="home-container"
>
  <div
    className="header-container"
    data-testid="header-container"
  >
    <div
      className="logo-title-container"
    >
      <img
        alt="logo"
        className="logo"
        data-testid="logo"
        src="logo.svg"
      />
      <div
        className="header-text"
      >
        Welcome to TSX
      </div>
    </div>
    <div
      className="sub-title"
    >
      Kulengaho
    </div>
  </div>
  <div>
    <div
      className="login-component-container"
      data-testid="login-component-container"
    >
      <div
        className="form-container"
      >
        <form
          data-testid="login-form"
          id="login-form"
          onSubmit={[Function]}
        >
          <div
            className="text-input-container"
          >
            <label>
              Username
            </label>
            <input
              className="text-input "
              data-testid="username-login"
              id="user-name"
              onChange={[Function]}
              placeholder="Username"
              type="string"
              value=""
            />
            
          </div>
          <div
            className="text-input-container"
          >
            <label>
              Password
            </label>
            <input
              className="text-input "
              data-testid="password-login"
              id="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            
          </div>
          <div
            className="button-container"
          >
            <button
              className="button"
              data-testid="login-submit"
              disabled={true}
              id="submit-button"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
